openfe/protocols/openmm_utils/omm_settings.py

```diff
@@ -29,38 +30,183 @@
+      * ``box_size``
+
+    Default settings will create a cubic box, although more space efficient
+    shapes (e.g. ``dodecahedrons``) are recommended to improve simulation
+    performance.
+
+
+    Notes
+    -----
+    * The number of water molecules added will be affected by the number of
+      ions defined in SolventComponent. For example, the value of
+      ``number_of_solvent_molecules`` is the sum of the number of counterions
+      added and the number of water molecules added.
+    * Solvent addition does not account for any pre-existing waters explicitly
+      defined in the :class:`openfe.ChemicalSystem`. Any waters will be added
+      in addition to those pre-existing waters.
+    * No solvation will happen if a SolventComponent is not passed.
+
+
+    See Also
+    --------
+    :mod:`openmm.app.Modeller`
+    Base class for SolvationSettings objects
+    """
+    solvent_model: Literal['tip3p', 'spce', 'tip4pew', 'tip5p'] = 'tip3p'
+    """
+    Force field water model to use when solvating and defining the model
+    properties (e.g. adding virtual site particles).
+
+    Allowed values are; `tip3p`, `spce`, `tip4pew`, and `tip5p`.
+    """
+    solvent_padding: Optional[FloatQuantity['nanometer']] = 1.2 * unit.nanometer
+    """
+    Minimum distance from any solute bounding sphere to the edge of the box.
 
     Note
     ----
-    No solvation will happen if a SolventComponent is not passed.
+    * Cannot be defined alongside ``number_of_solvent_molecules``,
+      ``box_size``, or ``box_vectors``.
+    """
+    box_shape: Optional[Literal['cube', 'dodecahedron', 'octahedron']] = 'cube'
+    """
+    The shape of the periodic box to create.
 
+    Notes
+    -----
+    * Must be one of `cube`, `dodecahedron`, or `octahedron`.
+    * Cannot be defined alongside ``box_vectors`` or ``box_size``.
     """
-    solvent_model: Literal['tip3p', 'spce', 'tip4pew', 'tip5p'] = 'tip3p'
+    number_of_solvent_molecules: Optional[int] = None
     """
-    Force field water model to use.
-    Allowed values are; `tip3p`, `spce`, `tip4pew`, and `tip5p`.
+    The number of solvent molecules (water + ions) to add.
+
+    Note
+    ----
+    * Cannot be defined alongside ``solvent_padding``, ``box_size``,
+      or ``box_vectors``.
+    """
+    box_vectors: Optional[ArrayQuantity['nanometer']] = None
     """
+    `OpenMM reduced form box vectors <http://docs.openmm.org/latest/userguide/theory/05_other_features.html#periodic-boundary-conditions>`.
 
-    solvent_padding: FloatQuantity['nanometer'] = 1.2 * unit.nanometer
-    """Minimum distance from any solute atoms to the solvent box edge."""
+    Notes
+    -----
+    * Cannot be defined alongside ``solvent_padding``,
+      ``number_of_solvent_molecules``, or ``box_size``.
+
+    See Also
+    --------
+    :mod:`openff.interchange.components.interchange`
+    :mod:`openff.interchange.components._packmol`
+    """
+    box_size: Optional[ArrayQuantity['nanometer']] = None
```

Comment:
This inherits the name from OpenMM, it's not amazingly named.

I.e. if you know what it is in OpenMM this makes sense, but if you don't then it may not...